src/zabapgit_object_shma.prog.abap

```diff
@@ -0,0 +1,285 @@
+          EXPORTING
+            mode_shma_attributes = 'E'
+            area_name            = area_name
+            x_area_name          = ' '
+            _scope               = '2'
+            _wait                = ' '
+            _collect             = ' '
+          EXCEPTIONS
+            foreign_lock         = 1
+            system_failure       = 2
+            OTHERS               = 3.
+
+        IF sy-subrc <> 0.
+          lcx_exception=>raise( |Error deleting SHMA { ms_item-obj_name }| ).
+        ENDIF.
+
+        CALL METHOD ('\PROGRAM=SAPMSHM_MONITOR\CLASS=LCL_SHMM')=>('FREE_AREA_BY_NAME')
+          EXPORTING
+            area_name     = area_name
+            affect_server = cl_shm_area=>affect_all_servers.
+
+        CREATE OBJECT lo_cts_if TYPE ('\FUNCTION-POOL=SHMA\CLASS=LCL_CTS_INTERFACE')
+          EXPORTING
+            area = area_name.
+
+        CALL METHOD lo_cts_if->('CHECK_AREA')
+          EXPORTING
+            request     = c_request_delete
+          IMPORTING
+            access_mode = request
+            appendable  = append.
+
+        IF request <> c_request_delete.
+          lcx_exception=>raise( |Error deleting SHMA { ms_item-obj_name }| ).
+        ENDIF.
+
+        CALL METHOD lo_cts_if->('INSERT_AREA')
+          EXPORTING
+            request = c_request_delete
+          IMPORTING
+            order   = order
+            task    = task.
+
+        DELETE FROM shma_attributes  WHERE area_name = area_name.
+        DELETE FROM shma_start       WHERE area_name = area_name.
+
+        korrnum = order.
+        objname = area_name.
+
+        CALL FUNCTION 'TR_TADIR_INTERFACE'
+          EXPORTING
+            wi_read_only      = abap_true
+            wi_tadir_pgmid    = 'R3TR'
+            wi_tadir_object   = 'SHMA'
+            wi_tadir_obj_name = objname
+          IMPORTING
+            new_tadir_entry   = tadir
+          EXCEPTIONS
+            OTHERS            = 0.
+
+        CALL FUNCTION 'TR_DEVCLASS_GET'
+          EXPORTING
+            iv_devclass = tadir-devclass
+          IMPORTING
+            es_tdevc    = tdevc
+          EXCEPTIONS
+            OTHERS      = 1.
+
+        IF  sy-subrc = 0 AND tdevc-korrflag IS INITIAL.
+
+          " TADIR entries for local objects must be deleted 'by hand'
+
+          CALL FUNCTION 'TR_TADIR_INTERFACE'
+            EXPORTING
+              wi_test_modus         = abap_false
+              wi_delete_tadir_entry = abap_true
+              wi_tadir_pgmid        = 'R3TR'
+              wi_tadir_object       = 'SHMA'
+              wi_tadir_obj_name     = objname
+              wi_tadir_korrnum      = korrnum
+            EXCEPTIONS
+              OTHERS                = 0.
+
+          CALL FUNCTION 'TR_TADIR_INTERFACE'
+            EXPORTING
+              wi_test_modus         = abap_false
+              wi_delete_tadir_entry = abap_true
+              wi_tadir_pgmid        = 'R3TR'
+              wi_tadir_object       = 'CLAS'
```

Comment:
hmm, how does this work? I guess the CLAS serialization should take care of the CLAS